### PR TITLE
feat(core): ldml C++ vkey support 🙀

### PR DESCRIPTION
**Pull request by srl295**
- add a vkey test - #7135

@keymanapp-test-bot skip

### CODE REVIEW

File: core/tests/unit/ldml/keyboards/k_103_vkey.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+@@keys: [K_A][K_U][K_Q][K_C][K_K][K_SLASH][K_Z][K_Q][K_L][K_T][K_W]
```
Comment (srl295) [inline core/tests/unit/ldml/keyboards/k_103_vkey.xml:4]: @mcdurdin  is this right?  AUQCK/ZQLTW -> quack/waltz

maybe this is partly a test question. do we expect process_key to 'honor' the vkey mappings at this level?

File: core/tests/unit/ldml/keyboards/k_103_vkey.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+@@expected: \u0071\u0075\u0061\u0063\u006b\u002f\u0077\u0061\u006c\u0074\u007a
+-->
+
+<!DOCTYPE keyboard SYSTEM "../../../../../resources/standards-data/ldml-keyboards/techpreview/dtd/ldmlKeyboard.dtd">
+<keyboard locale="en" conformsTo="techpreview">
+  <info author="srl295" indicator="🙀" layout="azerty" normalization="NFC" />
```
Comment (srl295) [inline core/tests/unit/ldml/keyboards/k_103_vkey.xml:10]: should `layout="azerty"` imply this vkey mapping?

File: core/tests/unit/ldml/keyboards/k_103_vkey.xml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  <names>
+    <name value="VKeyTestKbd" />
+  </names>
+
+  <vkeys>
+    <vkey from="Q" to="A" />
+    <vkey from="W" to="Z" />
+    <vkey from="A" to="Q" />
+    <vkey from="Z" to="W" />
+  </vkeys>
```
Comment (srl295) [inline core/tests/unit/ldml/keyboards/k_103_vkey.xml:21]: good candidate for an <import>

File: core/tests/unit/ldml/keyboards/k_103_vkey.xml (hunk 2 above)
Comment (mcdurdin) [reply core/tests/unit/ldml/keyboards/k_103_vkey.xml:10]: `layout="azerty"` is purely informational.

File: core/tests/unit/ldml/keyboards/k_103_vkey.xml (hunk 1 above)
Comment (mcdurdin) [reply core/tests/unit/ldml/keyboards/k_103_vkey.xml:4]: Yes, I think these test key events should be based on the US ANSI vk codes.

File: core/tests/unit/ldml/keyboards/k_103_vkey.xml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  <keys>
+    <import base="cldr" path="techpreview/keys-Zyyy-punctuation.xml"/>
+  </keys>
+
+  <layers form="hardware" hardware="us">
+    <layer id="base" modifier="none">
+      <row keys="grave 1 2 3 4 5 6 7 8 9 0 hyphen equal" />
+      <row keys="q w e r t y u i o p open-square close-square backslash" />
+      <row keys="a s d f g h j k l semi-colon apos" />
+      <row keys="z x c v b n m comma period slash" />
```
Comment (mcdurdin) [inline core/tests/unit/ldml/keyboards/k_103_vkey.xml:32]: This shows the layout as qwerty but the vkeys are remapped to azerty. Is this intentional? If so, we should add a comment.

Comment (srl295) [reply core/tests/unit/ldml/keyboards/k_103_vkey.xml:32]: > `a z e r t y`…

File: core/tests/unit/ldml/keyboards/k_103_vkey.xml (hunk 3 above)
Comment (srl295) [reply core/tests/unit/ldml/keyboards/k_103_vkey.xml:21]: Actually need a new Core API to test this.

**Pull request closed without merging**